browser-client.cpp

```diff
@@ -118,19 +118,80 @@ bool BrowserClient::OnProcessMessageReceived(
 			obs_frontend_start_virtualcam();
 		} else if (name == "stopVirtualcam") {
 			obs_frontend_stop_virtualcam();
+		} else if (name == "setProfile") {
+			obs_frontend_set_current_profile(
+				arguments->GetString(1).ToString().c_str());
+		} else if (name == "setSceneCollection") {
+			obs_frontend_set_current_scene_collection(
+				arguments->GetString(1).ToString().c_str());
 		}
 	case ControlLevel::Advanced:
 		if (name == "startReplayBuffer") {
 			obs_frontend_replay_buffer_start();
 		} else if (name == "stopReplayBuffer") {
 			obs_frontend_replay_buffer_stop();
+		} else if (name == "setScene") {
+			const std::string scene_name =
+				arguments->GetString(1).ToString();
+			obs_source_t *source =
+				obs_get_source_by_name(scene_name.c_str());
+			if (!source) {
+				blog(LOG_WARNING,
+				     "Browser source '%s' tried to switch to scene '%s' which doesn't exist",
+				     obs_source_get_name(bs->source),
+				     scene_name.c_str());
+			} else if (!obs_source_is_scene(source)) {
+				blog(LOG_WARNING,
+				     "Browser source '%s' tried to switch to '%s' which isn't a scene",
+				     obs_source_get_name(bs->source),
+				     scene_name.c_str());
+				obs_source_release(source);
+			} else {
+				obs_frontend_set_current_scene(source);
+				obs_source_release(source);
+			}
+		} else if (name == "setTransition") {
+			const std::string transition_name =
+				arguments->GetString(1).ToString();
+			obs_source_t *transition = obs_get_transition_by_name(
+				transition_name.c_str());
+			if (transition) {
+				obs_frontend_set_current_transition(transition);
+				obs_source_release(transition);
+			} else {
+				blog(LOG_WARNING,
+				     "Browser source '%s' tried to change the current transition to '%s' which doesn't exist",
+				     obs_source_get_name(bs->source),
+				     transition_name.c_str());
+			}
 		}
 	case ControlLevel::Basic:
 		if (name == "saveReplayBuffer") {
 			obs_frontend_replay_buffer_save();
 		}
 	case ControlLevel::ReadOnly:
-		if (name == "getCurrentScene") {
+		if (name == "getStatus") {
+			json = Json::object{
+				{"recording", obs_frontend_recording_active()},
+				{"streaming", obs_frontend_streaming_active()},
+				{"recordingPaused",
+				 obs_frontend_recording_paused()},
+				{"replaybuffer",
+				 obs_frontend_replay_buffer_active()},
+				{"virtualcam",
+				 obs_frontend_virtualcam_active()}};
+		} else if (name == "getScenes") {
+			struct obs_frontend_source_list list = {{{0}}};
+			obs_frontend_get_scenes(&list);
+			std::vector<char *> scenes_vector;
+			for (size_t i = 0; i < list.sources.num; i++) {
+				obs_source_t *source = list.sources.array[i];
+				scenes_vector.push_back(
+					(char *)obs_source_get_name(source));
+			}
+			json = Json::array{scenes_vector};
```

Comment:
I tested this just to make sure that this would function as it is written, and unfortunately it won't. Rather than an array of strings, it'll create an array with one item containing an array of strings. So basically it's currently creating something like this:
```json
[["string1", "string2", "string3"]]
```
which probably isn't what you wanted. I'm glad I checked this to verify.

Use
```cpp
json = scenes_vector
```
instead, which fixes it.